packages/nx-plugin/src/executors/verify-deps/executor.ts

```diff
@@ -0,0 +1,110 @@
+import { ExecutorContext, readJsonFile, writeJsonFile } from '@nx/devkit';
+import * as tmp from 'tmp';
+import { copySync } from 'fs-extra';
+import * as path from 'path';
+import * as semver from 'semver';
+import { execSync } from 'child_process';
+import { VerifyDepsExecutorSchema } from './schema';
+import { PackagePaths, getPackagePaths } from '../../utils';
+
+export default async function runExecutor(
+  options: VerifyDepsExecutorSchema,
+  context: ExecutorContext
+) {
+  const { cleanup: shouldCleanup } = options;
+  const { projectName, projectsConfigurations, root: workspaceRoot } = context;
+  if (!projectsConfigurations || !projectName) {
+    throw new Error('no project configurations');
+  }
+
+  const { root } = projectsConfigurations.projects[projectName];
+  const { name: tmpDir, removeCallback: cleanup } = tmp.dirSync({
+    prefix: projectName,
+    unsafeCleanup: true,
+  });
+  console.info('copying', projectName, 'to', tmpDir);
```

Comment:
I also suggest to do an early return if a package does not have `peerDependencies`.